core/locale/application/date_time_service_test.go

```diff
@@ -0,0 +1,133 @@
+package application
+
+import (
+	"flamingo.me/flamingo/v3/core/locale/domain"
+	"testing"
+	"time"
+
+	"flamingo.me/flamingo/v3/framework/flamingo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDateTimeService_GetDateTimeFormatterFromIsoString(t *testing.T) {
+	start := time.Now()
+	now, err := time.Parse(time.RFC3339, start.Format(time.RFC3339))
+	assert.NoError(t, err, "no error received")
+
+	// error for invalid time
+	formatter, err := new(DateTimeService).GetDateTimeFormatterFromIsoString("invalid time")
+	assert.Nil(t, formatter, "got nil for formatter")
+	assert.Error(t, err, "error received")
+
+	// error for invalid location
+	dateTimeService := DateTimeService{}
+	dateTimeService.Inject(flamingo.NullLogger{}, &struct {
+		DateFormat     string `inject:"config:core.locale.date.dateFormat"`
+		TimeFormat     string `inject:"config:core.locale.date.timeFormat"`
+		DateTimeFormat string `inject:"config:core.locale.date.dateTimeFormat"`
+		Location       string `inject:"config:core.locale.date.location"`
+	}{
+		Location: "invalid location",
+	})
+
+	formatter, err = dateTimeService.GetDateTimeFormatterFromIsoString(now.Format(time.RFC3339))
```

Comment:
Please do seperated subtest for each use case that improves readability of the test